mat-process-api/V1/Controllers/ProcessDataController.cs

```diff
@@ -54,21 +58,40 @@ public IActionResult GetProcessData(string processRef)
         /// If processData needs to be update, it's instead replaced with a full new one.
         /// Returning flat out 200 on successful update
         /// </summary>
-        /// <param name="processDataJSON"></param>
+        /// <param name="updateRequest"></param>
         /// <returns></returns>
-        [HttpPatch]
+        [HttpPut]
         [Produces("application/json")]
         [ProducesResponseType(StatusCodes.Status200OK)]
-        public IActionResult UpdateExistingProcessDocument()
+        public IActionResult UpdateExistingProcessDocument([FromBody]UpdateProcessDataRequest updateRequest)
         {
-            return Ok();
+            var isRequestValid = _updateValidator.Validate(updateRequest);
+            if (isRequestValid.IsValid)
+            {
+
+                try
+                {
+                    _logger.LogInformation($"Update ProcessData request for process ID {updateRequest.processDataToUpdate.Id}");
+                    var result = _processDataUsecase.ExecuteUpdate(updateRequest);
+                    return Ok(result);
+                }
+                catch (DocumentNotFound ex)
+                {
+                    return StatusCode(200, $"Document with reference {updateRequest.processDataToUpdate.Id} was not found in the database." +
+                        $" An update is not possible on non-existent documents.");
+                }
+                catch (Exception ex)
+                {
+                    return StatusCode(500, "An error has occured while processing the request - " + ex.Message + " " + ex.InnerException);
+                }
+            }
+            return BadRequest(isRequestValid.Errors);
         }
 
         /// <summary>
         /// Creates an intial process JSON document in the database.
         /// Upon creating a resource returns 200 Ok
         /// </summary>
-        /// <param name="processObject"></param>
```

Comment:
I think we need to provide some sort of parameter name for the sake of swagger documentation.

Reply:
I still feel that the POST endpoint should have a name for an input parameter specified in Swagger documentation. I don't think it's okay to just remove it because it's name matches another endpoints parameter name. But we can leave this for refactoring.